### PR TITLE
Migration: New porting guide from Cloud team

### PR DESCRIPTION
**Pull request by theotherjimmy**
Original text from @jen3andruska:
> New porting guide for porting Mbed OS before porting Mbed Cloud

### CODE REVIEW

**Comment by theotherjimmy** [comment]
I can at least rebase/filter-branch out the directory capitalization for you

**Comment by theotherjimmy** [comment]
Got it. Now the dir names match convention on this repo.

**Comment by ashok-rao** [comment]
Thanks @theotherjimmy ..
@iriark01 @AnotherButler : How do we proceed with this now?

**Comment by theotherjimmy** [comment]
Alright, I have updated the engineering content, removed passive vioce where possible and re-did the formatting to match other documents in this repo. Now: how can we merge this with the other tutorial in this repo.

**Comment by iriark01** [comment]
I suggest we wait for Amanda. She's back on Monday.

Thank you for all your help!

**Comment by ashok-rao** [comment]
@AnotherButler .. ping (hoping you are back in office)!

**Comment by AnotherButler** [comment]
@ashok-rao I'm reviewing this, and then I may move it into our porting section. I hope to have my edits complete tomorrow morning my time.

File: docs/tutorials/porting.md [hunk 1]
```diff
@@ -0,0 +1,225 @@
+## Mbed OS porting guide
```
Comment (AnotherButler) [inline docs/tutorials/porting.md]: I don't see any of the images on this page. Do we want to include them here or delete them?

File: docs/tutorials/porting.md [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+
+- The name of the target for builds is `MY_BOARD_1`.
+- The vendor is "MY_VENDOR", the device family is "MY_FAMILY" and the device on the board is "MY_DEVICE".
+- A Cortex-M4F based core.
+- Porting for Mbed OS v5.x, which requires support for Arm Compiler 5, GCC Arm Embedded and IAR EWARM.
+- Has a serial (UART) interface connected to the CMSIS-DAP implementation.
+- Requires the macro `CPU_DEVICE_1` to compile the vendor-provided HAL for the device mounted on the board.
+- Requires the macro `FAMILY_MY_FAMILY` to compile the vendor-provided HAL for the family of the device.
+
+All of these requirements directly map to relevant tags in the `targets.json` entry for our target. This is shown in step 3 below.
```
Comment (AnotherButler) [inline docs/tutorials/porting.md]: Note: We don't have a step 3 anymore. What does this reference?

File: docs/tutorials/porting.md (hunk 1 above)
Comment (theotherjimmy) [reply docs/tutorials/porting.md]: The last commit replaced all of them with the single tree structure at the bottom of the document.

File: docs/tutorials/porting.md (hunk 2 above)
Comment (theotherjimmy) [reply docs/tutorials/porting.md]: The following step.

**Comment by theotherjimmy** [comment]
@AnotherButler I think we can remove the "Amanda is on holiday. please hold." label (which I incidentally find funny).

**Comment by iriark01** [comment]
Oh I put that there...

**Comment by AnotherButler** [comment]
Closing in favor of #625

**Pull request closed without merging**